src/lib/tls/tls_server.cpp

```diff
@@ -231,7 +231,10 @@ Server::Server(const Callbacks& callbacks,
 Handshake_State* Server::new_handshake_state(Handshake_IO* io)
    {
    std::unique_ptr<Handshake_State> state(
-      new Server_Handshake_State(io, get_callbacks().handshake_msg()));
+      new Server_Handshake_State(io,
+                                 std::bind(&TLS::Callbacks::handshake_msg,
```

Comment:
We probably don't want to keep this bind here in the future. It would be a lot more readable and probably also more performant to just pass a Reference to TLS::Callbacks here.